### PR TITLE
mockgcp: Support for KMSKeyRing

### PR DESCRIPTION
**Pull request by justinsb**
- tests: capture kmskeyring golden output
- mockgcp: generate kms grpc/proto code
- mockgcp: Add initial support for KMSKeyRing

### CODE REVIEW

**Comment by justinsb** [comment]
cc @jingyih it looks like you're going to want mockgcp cloudkms support.  Have you started?  If not, I'm happy to do it...

**Comment by justinsb** [comment]
@jingyih shall we get this in, to get more test coverage on the vertexai resources you added?

**Comment by justinsb** [comment]
/assign @yuwenma 

I did this a while ago, I was going through my older PRs and this one still seems to be in good shape.  Shall we get it in?

File: mockgcp/mockkms/instance.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	r.populateDefaultsForKeyRing(name, obj)
+
+	if err := r.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	return obj, nil
+}
+
+func (r *kmsServer) populateDefaultsForKeyRing(name *KeyRingName, obj *pb.KeyRing) {
```
Comment (yuwenma) [inline mockgcp/mockkms/instance.go:80]: clean up?

**Comment by justinsb** [comment]
/hold cancel

I'll send another PR to clean it up :-)

**Pull request merged**